src/json.cpp

```diff
@@ -1072,7 +1073,7 @@ static bool get_escaped_or_unicode( std::istream &stream, std::string &s, std::s
                         }
                     }
                     try {
-                        s += utf16_to_utf8( u );
+                        s += utf32_to_utf8( u );
                     } catch( const std::exception &e ) {
```

Comment:
`utf32_to_utf8` does not throw when encountering an invalid unicode codepoint so I suggest removing this try-catch block and testing whether `u` is a valid unicode codepoint to preserve the original behavior.

Reply:
Well spotted; thanks!  Will fix that.

Reply:
Upon closer examination, the value `u` is computed from 4 hex characters, so it could never be large enough to hit that particular failure case anyway.

Non-BMP characters are supposed tp be encoded as a surrogate pair, so it turns out we're not handling those correctly, but that was already the case and I'm claiming it is out of scope for this PR.